fennel/internal_lib/to_proto/source_code.py

```diff
@@ -210,11 +282,96 @@ def lambda_to_python_regular_func(lambda_func):
+        get_text_between_ast_pointers(
+            source_text,
+            lambda_node.lineno,
+            lambda_node.col_offset,
+            (
+                lambda_node.end_lineno
+                if lambda_node.end_lineno
+                else lambda_node.lineno
+            ),
+            (
+                lambda_node.end_col_offset
+                if lambda_node.end_col_offset
+                else lambda_node.col_offset
+            ),
+        )
     )
+    lambda_next_line(lambda_func, lambda_text, line_num)
     return lambda_text
 
 
+def lambda_next_line(lambda_func: Callable, lambda_text: str, line_num: int):
+    """
+    This function checks if next_line after line_num can be included in the lambda or not.
+    If yes then throw an error if not then return
+    """
+    # Get the next line and append to lambda text
+    next_line = ""
+    with open(lambda_func.__code__.co_filename, "r") as f:
+        lines = f.readlines()
+        if line_num < len(lines):
+            next_line = lines[line_num].strip()
+    # No Next line found
+    if next_line == "":
+        return
+    # Next line already in lambda_text
+    if next_line in lambda_text:
+        return
+    # If Next line is part on next operator
+    if (
+        ".groupby(" in next_line
+        or ".filter(" in next_line
+        or ".assign(" in next_line
+        or ".transform(" in next_line
+        or ".rename(" in next_line
+        or ".select(" in next_line
+        or ".drop(" in next_line
+        or ".first(" in next_line
+        or ".latest(" in next_line
+        or ".aggregate(" in next_line
+        or ".join(" in next_line
+    ):
+        return
+    source_text = lambda_text.strip() + next_line
+    try:
+        source_ast = ast.parse(source_text)
+        lambda_node = next(
+            (
+                node
+                for node in ast.walk(source_ast)
+                if isinstance(node, ast.Lambda)
+            ),
+            None,
+        )
+        if lambda_node is not None:
+            new_lambda_text = dedent(
+                get_text_between_ast_pointers(
+                    source_text,
+                    lambda_node.lineno,
+                    lambda_node.col_offset,
+                    (
+                        lambda_node.end_lineno
+                        if lambda_node.end_lineno
+                        else lambda_node.lineno
+                    ),
+                    (
+                        lambda_node.end_col_offset
+                        if lambda_node.end_col_offset
+                        else lambda_node.col_offset
+                    ),
+                )
+            )
+            if new_lambda_text != lambda_text:
+                raise ValueError(
+                    f"Not able to choose lambda function between {lambda_text} and {new_lambda_text}. "
+                    f"If lambda contains more than one line, please keep in one () bracket."
```

Comment:
please define the function within the parentheses

Reply:
changed